templates/ui-ingress.yaml

```diff
@@ -1,7 +1,11 @@
 {{- if (and (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) (or (and (ne (.Values.ui.enabled | toString) "-") .Values.ui.enabled) (and (eq (.Values.ui.enabled | toString) "-") .Values.global.enabled)) (or (and (ne (.Values.ui.service.enabled | toString) "-") .Values.ui.service.enabled) (and (eq (.Values.ui.service.enabled | toString) "-") .Values.global.enabled))) }}
 {{- if (and (ne (.Values.ui.ingress.enabled | toString) "-") .Values.ui.ingress.enabled) }}
 {{- $serviceName := printf "%s-%s" (include "consul.fullname" .) "ui" -}}
-apiVersion: extensions/v1
+{{- if or ( gt .Capabilities.KubeVersion.Major "1" ) ( ge .Capabilities.KubeVersion.Minor "19" ) }}
```

Comment:
can you comment why we're not using `Capabilities.APIVersions.Has` using `{{- /* ..docs... */}}`

Reply:
Cool. I tested this out on 1.16 and 1.19 and it worked in both cluster. Need an annotation for `nginx` in `1.19` but that was straightforward. Will document the need for the above.

@lkysow Can you run the bats locally? It was failing on my machine but it passes in CI, because of how `helm2`/`helm3` results are different between CI and my local setup.

Reply:
I ask because im not quite certain how to work around this other than explicitly adding a KubeVersion check in our bats as well which would be a new pattern in the tests.

Reply:
Comment added.